okio/jvm/src/main/java/okio/Buffer.kt

```diff
@@ -259,35 +245,11 @@ actual class Buffer : BufferedSource, BufferedSink, Cloneable, ByteChannel {
   }
 
   @Throws(EOFException::class)
-  override fun readByte(): Byte {
-    if (size == 0L) throw EOFException()
-
-    val segment = head!!
-    var pos = segment.pos
-    val limit = segment.limit
-
-    val data = segment.data
-    val b = data[pos++]
-    size -= 1L
-
-    if (pos == limit) {
-      head = segment.pop()
-      SegmentPool.recycle(segment)
-    } else {
-      segment.pos = pos
-    }
-
-    return b
-  }
+  override fun readByte(): Byte = commonReadByte()
 
   /** Returns the byte at `pos`.  */
   @JvmName("getByte")
-  operator fun get(pos: Long): Byte {
-    checkOffsetAndCount(size, pos, 1L)
-    seek(pos) { s, offset ->
-      return s!!.data[(s.pos + pos - offset).toInt()]
-    }
-  }
+  operator fun get(pos: Long): Byte = commonGet(pos)
```

Comment:
mmmmmm . . . . . . this is going to have performance costs in a place where I’ve been really anxious about performance costs. This one method gets called a lot parsing JSON, and we’re inner-loop enough that just another stack frame is substantial

I’d like to fix more of the problem by pushing more into common, so that `get` is just commonGet(). How far are we from that?

Reply:
... I think we can workaround that by making `commonXxxx` methods inline, and avoid calling them anywhere but from `actual` functions.